analyzer-core/src/main/java/ffb/analyzer/core/queue/PriorityQueueManager.java

```diff
@@ -0,0 +1,71 @@
+package ffb.analyzer.core.queue;
+
+import java.util.concurrent.PriorityBlockingQueue;
+
+public class PriorityQueueHandler{
+    private enum QueueState {
+        STOPPED,
+        RUNNING
+    }
+
+    private static final int MAX_CAPACITY = 1000;
+    private static final int STARTING_TASK_ID = 1;
+
+    private PriorityBlockingQueue<PriorityTask> queue;
+    private int nextAvailableId;
+    private QueueState state;
+
+    /**
+     * Adds entry to queue and updates tracking information within PriorityQueueHandler.
+     * @param entry Task object to be added to the queue.
+     * @return True if the entry was successfully added, otherwise False.
+     */
+    public boolean add(PriorityTask entry) {
```

Comment:
This should set the `id` for the task, so that they are in sync. Otherwise, `PriorityQueueManger` can't guarantee that the task that's going to be run is actually the highest priority.

Reply:
I now use the return value from getAndIncrementNextAvailableId to set the id.